Feeder audit 2026-05-29: waterinfo-vmm

## Feeder audit summary for `waterinfo-vmm`

This sweep covers the objective repo-state parts of the feeder-release-checklist: required file presence, catalog + ghpages source entries, source-local pytest with generated producer src paths injected into PYTHONPATH, Docker E2E matrix coverage, ARM static audit, Fabric static audit, and ARM generator dry-run no-diff. It does not rerun live Azure/Fabric deploy bullets, specialist-agent reviews, or upstream API re-audits.

## Findings

- **ERROR** `missing-file` — 2 finding(s)
  - Required file missing: feeders/waterinfo-vmm/xreg/waterinfo-vmm.xreg.json (feeders/waterinfo-vmm/xreg/waterinfo-vmm.xreg.json)
  - Required file missing: feeders/waterinfo-vmm/kql/waterinfo-vmm.kql (feeders/waterinfo-vmm/kql/waterinfo-vmm.kql)
- **ERROR** `pytest-failed` — 1 finding(s)
  - Source-local pytest failed. (feeders/waterinfo-vmm/tests)
- **WARNING** `arm-stub-description` — 2 finding(s)
  - azure-template-with-eventhub.json: parameter 'eventHubSku' has stub or missing metadata.description (feeders/waterinfo-vmm/azure-template-with-eventhub.json)
  - azure-template-with-servicebus.json: parameter 'queueName' has stub or missing metadata.description (feeders/waterinfo-vmm/azure-template-with-servicebus.json)

## Check evidence

### `arm_audit` — rc=0, 1.41s

```text
ARM template static audit
  RepoRoot:        C:\git\real-time-sources
  Feeders scanned: 1
  Templates:       5
  Blockers:        0
  Warnings:        2

[waterinfo-vmm] blockers=0 warnings=2
  azure-template-with-eventhub.json
    [WARNING] stub-description: parameter 'eventHubSku' has stub or missing metadata.description
  azure-template-with-servicebus.json
    [WARNING] stub-description: parameter 'queueName' has stub or missing metadata.description
PASS (with 2 warning(s)).
Report written to C:\Users\clemensv\.copilot\session-state\cb5ea344-87f5-406e-a13a-a66811763074\files\fleet-release-audit\arm\waterinfo-vmm.json

[31;1mMethodInvocationException: [0mC:\git\real-time-sources\tools\validate-arm-templates.ps1:104[0m
[31;1m[0m[36;1mLine |[0m
[31;1m[0m[36;1m[36;1m 104 | [0m         [36;1m$matches = [regex]::Matches($text, '(?:os\.getenv|os\.environ[0m …[0m
[31;1m[0m[36;1m[36;1m[0m[36;1m[0m[36;1m     | [31;1m         ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~[0m
[31;1m[0m[36;1m[36;1m[0m[36;1m[0m[36;1m[31;1m[31;1m[36;1m     | [31;1mException calling "Matches" with "2" argument(s): "Value cannot be null. (Parameter 'input')"[0m
```

### `arm_generator` — rc=0, 3.68s

```text
Description sources: total=177, container=31, readme=0, argparse=28, fallback=118
Generated 0 ARM template(s).
```

### `fabric_audit` — rc=0, 1.31s

```text
[32;1mSource       [0m[32;1m OptIn[0m[32;1m Notebook[0m[32;1m  Hook[0m[32;1m PASS[0m[32;1m WARN[0m[32;1m BLOCK[0m
[32;1m------       [0m [32;1m-----[0m [32;1m--------[0m [32;1m ----[0m [32;1m----[0m [32;1m----[0m [32;1m-----[0m
waterinfo-vmm  True     True False   10    0     0

Sources: 1 total, 0 blockers, 0 warnings
JSON report: C:\Users\clemensv\.copilot\session-state\cb5ea344-87f5-406e-a13a-a66811763074\files\fleet-release-audit\fabric\waterinfo-vmm.json
```

### `pytest` — rc=1, 6.01s

```text
try:
            return self.version(api_version=False)["ApiVersion"]
        except KeyError as ke:
            raise DockerException(
                'Invalid response from docker daemon: key "ApiVersion"'
                ' is missing.'
            ) from ke
        except Exception as e:
>           raise DockerException(
                f'Error while fetching server API version: {e}'
            ) from e
E           docker.errors.DockerException: Error while fetching server API version: (2, 'CreateFile', 'The system cannot find the file specified.')

C:\Users\clemensv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\docker\api\client.py:230: DockerException
============================== warnings summary ===============================
..\..\..\..\Users\clemensv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\avro\schema.py:1229
  C:\Users\clemensv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\avro\schema.py:1229: IgnoredLogicalType: Logical type timestamp-millis requires literal type long, not string.
    warnings.warn(

..\..\..\..\Users\clemensv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\testcontainers\core\waiting_utils.py:215
  C:\Users\clemensv\AppData\Local\Packages\PythonSoftwareFoundation.Python.3.11_qbz5n2kfra8p0\LocalCache\local-packages\Python311\site-packages\testcontainers\core\waiting_utils.py:215: DeprecationWarning: The @wait_container_is_ready decorator is deprecated and will be removed in a future version. Use structured wait strategies instead: container.waiting_for(HttpWaitStrategy(8080).for_status_code(200)) or container.waiting_for(LogMessageWaitStrategy('ready'))
    @wait_container_is_ready()

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html

---------- coverage: platform win32, python 3.11.9-final-0 -----------
Name                             Stmts   Miss  Cover   Missing
--------------------------------------------------------------
waterinfo_vmm\__init__.py            0      0   100%
waterinfo_vmm\__main__.py            3      3     0%   2-5
waterinfo_vmm\waterinfo_vmm.py     184    107    42%   24, 31-34, 42-49, 72-75, 79-84, 88-94, 98-117, 166, 170, 187-188, 202-212, 219-297, 301
--------------------------------------------------------------
TOTAL                              187    110    41%
Coverage HTML written to dir htmlcov

=========================== short test summary info ===========================
FAILED tests/test_waterinfo_vmm_container.py::TestWaterinfoContainerIntegration::test_readings_delivered_to_kafka
!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!
1 failed, 1 passed, 2 warnings in 2.41s
```

## Next step

Fix the findings above, then rerun the sequential feeder-release-checklist audit for this feeder and close this issue with the new evidence.

--- a/feeders/waterinfo-vmm/tests/test_waterinfo_vmm_container.py
+++ b/feeders/waterinfo-vmm/tests/test_waterinfo_vmm_container.py
@@ -65,7 +65,7 @@ class TestWaterinfoContainerIntegration:
                 unit_name="meter",
                 parameter_name="H",
             )
-            waterinfo_producer.send_be_vlaanderen_waterinfo_vmm_water_level_reading(reading)
+            waterinfo_producer.send_be_vlaanderen_waterinfo_vmm_water_level_reading(reading.station_no, reading)
 
             # Send a mock station
             station = Station(
@@ -79,7 +79,7 @@ class TestWaterinfoContainerIntegration:
                 ts_id="306367042",
                 ts_unitname="meter",
             )
-            waterinfo_producer.send_be_vlaanderen_waterinfo_vmm_station(station)
+            waterinfo_producer.send_be_vlaanderen_waterinfo_vmm_station(station.station_no, station)
 
             # Consume and verify
             consumer = _create_consumer(bootstrap_servers)
@@ -136,7 +136,7 @@ class TestWaterinfoLiveContainerIntegration:
                     ts_id="",
                     ts_unitname="",
                 )
-                waterinfo_producer.send_be_vlaanderen_waterinfo_vmm_station(station, flush_producer=False)
+                waterinfo_producer.send_be_vlaanderen_waterinfo_vmm_station(station.station_no, station, flush_producer=False)
             producer.flush()
 
             consumer = _create_consumer(bootstrap_servers)
@@ -177,7 +177,7 @@ class TestWaterinfoLiveContainerIntegration:
                     unit_name=entry.get("ts_unitname", "meter"),
                     parameter_name=entry.get("stationparameter_name", "H"),
                 )
-                waterinfo_producer.send_be_vlaanderen_waterinfo_vmm_water_level_reading(reading, flush_producer=False)
+                waterinfo_producer.send_be_vlaanderen_waterinfo_vmm_water_level_reading(reading.station_no, reading, flush_producer=False)
                 sent_count += 1
             producer.flush()
 

